library/core/src/intrinsics/mod.rs

```diff
@@ -2823,6 +2825,7 @@ pub const fn offset_of<T: PointeeSized>(variant: u32, field: u32) -> usize;
 #[rustc_nounwind]
 #[unstable(feature = "core_intrinsics", issue = "none")]
 #[rustc_intrinsic]
+#[rustc_intrinsic_const_stable_indirect]
 pub const fn variant_count<T>() -> usize;
```

Comment:
This isn't actually stable either, I think? `std::mem::variant_count` is still unstable.